### PR TITLE
Notify engine about VKB state

### PR DESCRIPTION
**Pull request by rojkov**
This PR depends on xulrunner 31.3.0.4 and the commit
https://github.com/nemomobile-packages/embedlite-components/commit/1ffd093cbf31bf9bc90b603cdf2b74e3b7d08e20

### CODE REVIEW

File: src/declarativewebcontainer.cpp [hunk 1]
```diff
@@ -872,3 +875,39 @@ void DeclarativeWebContainer::sendVkbOpenCompositionMetrics()
         m_webPage->sendAsyncMessage("embedui:vkbOpenCompositionMetrics", data);
     }
 }
+
+void DeclarativeWebContainer::notifyVkbFullyOpen()
+{
+    int vkbHeight = 0;
+    int winHeight = 0;
+
+    if (width() < height()) {
```
Comment (rainemak) [inline src/declarativewebcontainer.cpp]: ```if (!m_portrait) {```

File: src/declarativewebcontainer.cpp [hunk 2]
```diff
@@ -872,3 +875,39 @@ void DeclarativeWebContainer::sendVkbOpenCompositionMetrics()
         m_webPage->sendAsyncMessage("embedui:vkbOpenCompositionMetrics", data);
     }
 }
+
+void DeclarativeWebContainer::notifyVkbFullyOpen()
+{
```
Comment (rainemak) [inline src/declarativewebcontainer.cpp]:
```
if (!window()) {
    return;
}
```
Not 100% sure if this is possible even in theory.

File: src/declarativewebcontainer.cpp [hunk 3]
```diff
@@ -77,6 +78,7 @@ DeclarativeWebContainer::DeclarativeWebContainer(QQuickItem *parent)
     connect(this, SIGNAL(heightChanged()), this, SLOT(sendVkbOpenCompositionMetrics()));
     connect(this, SIGNAL(widthChanged()), this, SLOT(sendVkbOpenCompositionMetrics()));
```
Comment (rainemak) [inline src/declarativewebcontainer.cpp]: Could we now remove these connections?

**Comment by rainemak** [comment]
Looks to me that sendVkbOpenCompositionMetrics calculates incorrectly

**Comment by rojkov** [comment]
I've addressed the comments and in addition to them I've dropped updateVkbHeight() and added orientation change handling.

The corresponding patch in embedlite-components is https://github.com/rojkov/embedlite-components/commit/e1636fe7ad38b9fc18e3651bee2a42da3c89e698

**Comment by rainemak** [comment]
Ran through some random web pages. Here are some observations:

- m_inputPanelOpenHeight should be reset to zero when input panel is focused as virtual keyboard size can vary (digits, landscape/portrait)
- setter of inputPanelHeight (read notify) could be removed from QML side. We could update actual target values immediately when QInputMethod::visible() changes (per orientation).
- If above approach would work, then WebView could binding directly to Qt.inputMethod.visible ~ WebView.qml@298
- regarding the zooming delay, we could calculate target viewport size when virtual keyboard starts opening (when im visibility changes). Send that information immediately to the engine side and handle fully opened state there

**Comment by rojkov** [comment]
I've dropped the inputPanel related properties. So far no regressions.

Sending vieport size right after VKB starts to open isn't better perfromance-wise:
1. currently metrics updates in AZPC are propagated synchronously to sailfish-browser C++ code without conversions
2. handling the updates in embedhepler.js would require data transformations C++ <-> JS.

**Comment by rojkov** [comment]
clean up open PRs

**Pull request closed without merging**